### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by James-Mart**
## Content changes
* I deleted the docs that describe how to generate wildcard certs with certbot, and use nginx/godaddy. Those instructions were too specific and could become outdated if those third-parties change and therefore they are outside the scope of our docs. I added a more general replacement in `run-infrastructure/configuration/https.md`.
* Rewrote much of the introduction page (`/README.md`), which was generally outdated and not very well written. 
* Added a brief page describing the intent of the AccountNumbers object (`/development/format/account-numbers.md`)

## Structural changes
* Eliminated all of the manual tables of contents, and added a table-of-contents plugin that adds TOC to the right hand side of the docs (in desktop view).
* Reorganized the entire book into categories that I think make more sense. There's a lot more work to be done in this category, but this is a start. 
* Reorganized the structure of the book to have README.md inside each section, which is the convention used by the mdbook docs themselves. Makes it easier to locate the intro page for each section.

## Mdbook changes
* Updated the version of mermaid used by the mdbook-mermaid crate by dynamically loading the mermaid library instead of bundling it with the book. The version bundled with mdbook-mermaid is old.
* The docs now support the "edit on github" button, which will link to the correct page on which edits are being made.
* Turned on `mdbook-linkcheck` by default to detect at build time if anyone breaks links in the docs
* Turned off the "print" button, which didn't work well for our book anyway with mermaid diagrams, etc.

## Further notes
* Lots of links don't have a corresponding page yet, I don't plan to resolve all missing pages in this PR.

### CODE REVIEW

**Comment by James-Mart** [comment]
Eliminated the doc on interfaces, which I will add in a separate PR

File: doc/psidk/src/development/services/README.md
Comment (swatanabe) [inline doc/psidk/src/development/services/README.md:1]: I don't think the low-level details of webassembly support belong on the same page as the high-level overview of services.

File: doc/psidk/src/development/services/simpleUI.md
Comment (swatanabe) [inline doc/psidk/src/development/services/simpleUI.md:1]: What is this file for? I don't see a corresponding html page or index entry.

File: doc/psidk/theme/js/mermaid-load.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const script = document.createElement('script');
+script.src = 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.min.js';
```
Comment (swatanabe) [inline doc/psidk/theme/js/mermaid-load.js:2]: I know I'm in a minority, but I like being able to view documentation offline...

File: doc/psidk/theme/css/pagetoc.css
Comment (swatanabe) [inline doc/psidk/theme/css/pagetoc.css:1]: The layout has a few issues (depending on the window size)
- The vertical next button can cover the page TOC making it hard to use it to navigate
- The size and position of the TOC don't seem to match, as the right side can overflow the window

File: doc/psidk/src/applet/applets.md
Comment (swatanabe) [inline doc/psidk/src/applet/applets.md:1]: I can find neither where this file was moved to nor an explanation of why it was deleted.

File: doc/psidk/src/applet/inter-applet-communication.md
Comment (swatanabe) [inline doc/psidk/src/applet/inter-applet-communication.md:1]: Why is this removed?

Comment (James-Mart) [reply doc/psidk/src/applet/inter-applet-communication.md:1]: It was largely rewritten and moved into the app interfaces document, which I then moved out of this PR.

File: doc/psidk/src/applet/applets.md
Comment (James-Mart) [reply doc/psidk/src/applet/applets.md:1]: It was mostly rewritten, and the information was split between multiple other documents. Most of which is now part of the app interfaces docs, which I moved out of this pr.

File: doc/psidk/src/development/services/README.md
Comment (James-Mart) [reply doc/psidk/src/development/services/README.md:1]: Done; Isolated them into a separate doc

File: doc/psidk/src/development/services/simpleUI.md
Comment (James-Mart) [reply doc/psidk/src/development/services/simpleUI.md:1]: Leftover from a prior iteration; deleted

File: doc/psidk/theme/css/pagetoc.css
Comment (James-Mart) [reply doc/psidk/theme/css/pagetoc.css:1]: * The next/previous buttons have been hidden. Not very useful controls anyways.

* The TOC is hidden on smaller resolution screens, and the docs look good on at least the following devices with known resolutions:
image

The only real issue is on desktops where the resolution is unusual or the window is not maximized. I think this won't affect many people and can be fixed later.

File: doc/psidk/theme/js/mermaid-load.js (hunk 1 above)
Comment (James-Mart) [reply doc/psidk/theme/js/mermaid-load.js:2]: You can, your penalty is just that you will have to deal with a text representation of the mermaid diagram 😜

File: doc/psidk/src/development/services/README.md
Comment (swatanabe) [reply doc/psidk/src/development/services/README.md:1]: error: It looks like "development/services/webassembly.md" wasn't included in SUMMARY.md

**Review by swatanabe — APPROVED** [review]
